fineract-provider/src/main/java/org/apache/fineract/portfolio/client/data/ClientData.java

```diff
@@ -24,92 +24,96 @@
 import java.util.List;
 import java.util.Set;
 import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.lang3.builder.CompareToBuilder;
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.apache.fineract.infrastructure.codes.data.CodeValueData;
 import org.apache.fineract.infrastructure.core.data.EnumOptionData;
+import org.apache.fineract.infrastructure.core.domain.ExternalId;
 import org.apache.fineract.infrastructure.dataqueries.data.DatatableData;
 import org.apache.fineract.organisation.office.data.OfficeData;
 import org.apache.fineract.organisation.staff.data.StaffData;
 import org.apache.fineract.portfolio.address.data.AddressData;
-import org.apache.fineract.portfolio.collateralmanagement.domain.ClientCollateralManagement;
 import org.apache.fineract.portfolio.group.data.GroupGeneralData;
 import org.apache.fineract.portfolio.savings.data.SavingsAccountData;
 import org.apache.fineract.portfolio.savings.data.SavingsProductData;
 
 /**
  * Immutable data object representing client data.
  */
+@NoArgsConstructor
 @Getter
+@Setter
 public final class ClientData implements Comparable<ClientData>, Serializable {
 
-    private final Long id;
-    private final String accountNo;
-    private final String externalId;
-
-    private final EnumOptionData status;
-    private final CodeValueData subStatus;
-
-    private final Boolean active;
-    private final LocalDate activationDate;
-
-    private final String firstname;
-    private final String middlename;
-    private final String lastname;
-    private final String fullname;
-    private final String displayName;
-    private final String mobileNo;
-    private final String emailAddress;
-    private final LocalDate dateOfBirth;
-    private final CodeValueData gender;
-    private final CodeValueData clientType;
-    private final CodeValueData clientClassification;
-    private final Boolean isStaff;
-
-    private final Long officeId;
-    private final String officeName;
-    private final Long transferToOfficeId;
-    private final String transferToOfficeName;
-
-    private final Long imageId;
-    private final Boolean imagePresent;
-    private final Long staffId;
-    private final String staffName;
-    private final ClientTimelineData timeline;
-
-    private final Long savingsProductId;
-    private final String savingsProductName;
-
-    private final Long savingsAccountId;
-    private final EnumOptionData legalForm;
-    private final Set<ClientCollateralManagementData> clientCollateralManagements;
+    private Long id;
+    private String accountNo;
+    private ExternalId externalId;
```

Comment:
Please add a manual getter for this, which return `ExternalId.empty()` if it is null.

Reply:
Done